onnxruntime/core/session/inference_session.cc

```diff
@@ -1078,6 +1079,19 @@ common::Status InferenceSession::TransformGraph(onnxruntime::Graph& graph, bool
     ORT_RETURN_IF_ERROR_SESSIONID_(apply_transformer_once(copy_transformer, *session_logger_, graph));
   }
 
+// Memory optimization graph.
+#ifdef ENABLE_TRAINING
+  {
+    const std::string enable_memory_optimizer =
```

Comment:
what is the expected behavior if memory config is empty?

One more thing - will be good to change this to a more intuitive name like memory_optimizer_config instead of enable_memory_optimizer. Current name gives an impression this acts like bool and simply enables or disables this feature.

Reply:
Thank you. Fixed.